### PR TITLE
chore: Bump OZ dependencies to 4.9.3 and add protocolVersion

### PR DESCRIPTION
**Pull request by Michael-A-Heuer**
## Description

Task: [OS-656](https://aragonassociation.atlassian.net/browse/OS-656)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] Any dependent changes have been merged and published in downstream modules.
- [x] I ran all tests with success and extended them if necessary.
- [x] I have updated the `CHANGELOG.md` file in the root folder.
- [x] I have updated the `DEPLOYMENT_CHECKLIST` file in the root folder.
- [x] I have updated the `UPDATE_CHECKLIST` file in the root folder.

[OS-656]: https://aragonassociation.atlassian.net/browse/OS-656?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/contracts/src/core/dao/DAO.sol [hunk 1]
```diff
@@ -69,9 +69,10 @@ contract DAO is
     /// @notice The second out of two values to which the `_reentrancyStatus` state variable (used by the `nonReentrant` modifier) can be set inidicating that a function was entered.
     uint256 private constant _ENTERED = 2;
 
-    /// @notice The [ERC-1271](https://eips.ethereum.org/EIPS/eip-1271) signature validator contract.
-    /// @dev Added in v1.0.0.
-    IERC1271 public signatureValidator;
+    /// @notice Removed variable that is left here to maintain the storage layout.
+    /// @dev Introducedd in v1.0.0. Removed in v1.4.0.
+    /// @custom:oz-renamed-from signatureValidator
+    address private __removed1;
```
Comment (Rekard0) [inline packages/contracts/src/core/dao/DAO.sol:75]: are we sure `__removed1` ?
i immediately start looking for `__removed0`

**Review by Rekard0 — COMMENTED** [review]
Have you or Could you confirm that those interfaces https://github.com/aragon/osx/blob/develop/packages/subgraph/src/utils/constants.ts#L26
didn't change?

File: packages/contracts/src/core/dao/DAO.sol (hunk 1 above)
Comment (Michael-A-Heuer) [reply packages/contracts/src/core/dao/DAO.sol]: True, renamed here c27d1c41873626f16f03a33b55871a754a76bf9a

**Review by mathewmeconry — DISMISSED** [review]

**Comment by Michael-A-Heuer** [comment]
> Have you or Could you confirm that those interfaces https://github.com/aragon/osx/blob/develop/packages/subgraph/src/utils/constants.ts#L26 didn't change?

We do this in https://aragonassociation.atlassian.net/browse/OS-694

File: packages/contracts/test/core/permission/permission-manager.ts [hunk 2]
```diff
@@ -20,10 +20,10 @@ const RESTRICTED_PERMISSIONS_FOR_ANY_ADDR = [
...
 
-const UNSET_FLAG = ethers.utils.getAddress(
+export const UNSET_FLAG = ethers.utils.getAddress(
   '0x0000000000000000000000000000000000000000'
 );
-const ALLOW_FLAG = ethers.utils.getAddress(
+export const ALLOW_FLAG = ethers.utils.getAddress(
   '0x0000000000000000000000000000000000000002'
 );
 export const ANY_ADDR = '0xffffffffffffffffffffffffffffffffffffffff';
```
Comment (Rekard0) [inline packages/contracts/test/core/permission/permission-manager.ts:29]: shouldn't these be in a global contast place?

**Review by Rekard0 — APPROVED** [review]

File: packages/contracts/test/core/permission/permission-manager.ts (hunk 2 above)
Comment (Michael-A-Heuer) [reply packages/contracts/test/core/permission/permission-manager.ts]: We don't have a global constants file for OSX as subgraph has. All the constants are scattered everywhere. 
Maybe this needs some general refactoring at some point, which might also be important for the code split.

Here, the export was even redundant so I removed it in e4f3fc28fc0b4e2d266863566e919ec6d8ec3741.

**Review by mathewmeconry — APPROVED** [review]

**Pull request merged**